models/core/revenuecat_subscription_transactions.sql

```diff
@@ -50,11 +50,58 @@ renamed as (
         price_in_usd * tax_percentage as estimated_tax_in_usd,
         price_in_usd - commission_in_usd - estimated_tax_in_usd as proceeds_in_usd,
         -- Documentation for normalizing mrr_in_usd: https://docs.revenuecat.com/docs/monthly-recurring-revenue-mrr_in_usd-chart
-        case
-            when product_duration = 'P1M' then price_in_usd
-            when product_duration = 'P1Y' then price_in_usd / 12
-            when product_duration = 'P1W' then price_in_usd * 4
-            else 0
+		case when effective_end_time is not null then
+            case 
+                /* handle cases where product_duration cannot be used for the transaction first */
+                when (is_in_intro_offer_period = 'true' or product_duration is null) then 
+                case
+                    when datediff(day, start_time, expected_end_time) between 0 and 1 
+                        then (30 * price_in_usd)::decimal(18,2)
```

Comment:
Can we use dbt.safe_cast? Its preferred for hub packages to avoid using overly dw specific functionality

Reply:
I removed the casting completely it wasn't necessary anyways this was just a copy+paste from the RevenueCat documentation

Reply:
It actually did matter quite a bit. I changed it to use round() which is accepted by most datawarehouses